lib/opensusebasetest.pm

```diff
@@ -152,8 +152,13 @@ sub investigate_yast2_failure {
     );
     for my $y2log_error (@y2log_errors) {
         if (my $y2log_error_result = script_output 'grep -B 3 "' . $y2log_error . '" /var/log/YaST2/y2log | tail -n 20 || true') {
-            record_info 'YaST2 log error detected', "Details:\n\n$y2log_error_result", result => 'fail';
-            $error_detected = 1;
+            if ($y2log_error eq '<3>' && $y2log_error_result =~ /not mounted/) {
```

Comment:
This will work, but we might expect more failures we will need to soft-fail.
What I would do is I would convert or extend @y2log_errors to the map, where one can add bug reference for known issues, so if we detect some failure and bug reference is defined, we will soft-fail, otherwise hard-fail. This will simplify code in the loop, so next time we just need to define data for the failure. WDYT?

Reply:
I would like that

Reply:
I had the same en lighting :) at the beginning but  I didn't know what to do to avoid missing ret code <3>. I remembered that the original idea was to capture individual error to have better separate bug report for developers in the way that I got to the conclusion that what we are doing is to detect ONLY what we expect, so hard fail in the generic <3> doesn't make sense, so I decided to take note of those errors with the surprise that we had a `tail -n 20` written since a lot of time ago hiding most of the 252 errors in logs with <3>. I run this to see if I could match them all and was successful:
```
grep -Ev "<3>.*no[t]? mount|<3>.*Error output: dracut:|<3>.*Reading install.inf|<3>.*shellcommand|<3>.*libstorage.*device not found|<3>.*lib/cheetah.rb.*Error output|<3>.*Slides.rb.*Directory.*does not exist|<3>.*agent-ini.*Can not open|<3>.*agent-ini.*Unable to stat|<3>.*Interpreter.*File not found|<3>.*Interpreter.*Couldn't find an agent|<3>.*Interpreter.*Read.*failed|<3>.*ag_uid.*argument is not a path|<3>.*ag_uid.*wrong command|<3>.*Interpreter.*'Syslog' failed|<3>.*libycp.*No matching component found|<3>.*Perl.*Perl call of Log|<3>.*Y2Ruby.*SSHAuthorizedKeys.write_keys failed" ~/Downloads/y2log | grep -E "<3>"
```
PR and VR are updated with those changes. Let me know if it is more useful in that way.